x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```diff
@@ -199,10 +210,47 @@ public InferencePipelineAggregationBuilder rewrite(QueryRewriteContext context)
                     delegate.onFailure(LicenseUtils.newComplianceException(XPackField.MACHINE_LEARNING));
                 }
             }));
+
+
+        context.registerAsyncAction((client, listener) -> {
+            if (licenseState.isSecurityEnabled()) {
+                // check the user has ml privileges
+                SecurityContext securityContext =new SecurityContext(Settings.EMPTY, client.threadPool().getThreadContext());
+                useSecondaryAuthIfAvailable(securityContext, () -> {
+                    final String username = securityContext.getUser().principal();
+                    final HasPrivilegesRequest privRequest = new HasPrivilegesRequest();
+                    privRequest.username(username);
+                    privRequest.clusterPrivileges("manage_ml", "monitor_ml");
```

Comment:
Does this stop really high privilege users from using inference?  For example, the actions permitted by `manage_ml` are a subset of those permitted by `manage` (which means manage everything except security), but literally checking for `manage_ml` might not return `true` for a user who has `manage`.  I may be wrong as I haven't tested it, but it seems likely that the security code will resolve high level cluster privilege names to sets of actions they allow, but not to other high level cluster privilege names.

You could check for just `GetTrainedModelsAction.NAME` here instead to solve that potential problem and also make the later code simpler as you'll only be testing for one privilege:

```suggestion
                    privRequest.clusterPrivileges(GetTrainedModelsAction.NAME);
```

Reply:
Yes that works